Monika After Story/game/definitions.rpy

```diff
@@ -5500,13 +5500,25 @@ init -1 python:
 
         RETURNS: True if a proccess in proc_list is running, False otherwise
         """
-        running_procs = get_procs()
-        if len(running_procs) == 0:
-            return False
-
-        for proc in proc_list:
-            if proc in running_procs:
-                return True
+        if renpy.linux:
+            for proc in proc_list:
+                try:
+                    output = subprocess.call(["pidof", proc])
+
+                    # 0: Success
+                    # 1: Did not find
+                    if output == 0:
+                        return True
```

Comment:
Looking on this again, may have found a more cross-platform working solution. The following *should* work on both Linux and MacOS:
```suggestion
                    output = re.search(
                        r"\/?{}(\s+--?\w+)*".format(proc),
                        bytes.decode(subprocess.check_output(["ps", "-A"])),
                        re.IGNORECASE
                    )

                    # not None means we found a match
                    if output is not None:
                        return True
```

Reply:
Might be helpful to add a comment on the Regex used on line 5507.

Also, this might not work for Linux; the outputs between `ps -A` on macOS and Linux vary a fair bit as you can see here:

![image](https://user-images.githubusercontent.com/40577357/162853895-7317181b-7eeb-4231-8927-f2b5c9b10228.png)

In particular, I'd like to note the following things:

- macOS' `ps` shows paths and arguments. Linux's `ps` does not show that; instead it shows the "title" of the process, not the actual process name.
- Atleast on my Arch Linux install, `ps` is from `procps-ng`. macOS doesnt even show a version, which is odd. They're definitely differing versions.

Reply:
Oh I noticed, I tested against it, however when I tested on Catalina at least, was possible to at least catch things i.e. Finder/Safari, though do note I didn't try much further. Also caught `trustd` as a test.

Also backup tested on Manjaro which also seeemed to work fine. It does bug the absolute hell outta me that Mac's is so...painful lol

Reply:
That said, was going to run final tests on there if this went in. (Mac install is laggy af w/o any GPU passthru)

Reply:
Yeah, if anything we can probably leave a `#NOTE:` on it mentioning it could potentially have issues on Mac due to a more verbose proc name.

Reply:
Hmm, the more I think about it the more I feel it would be best if we left this implementation for macOS use only, since the problem specifically is that Linux is the odd one out, not macOS.

This implementation will work great with macOS, but factoring in Linux with it starts making it a bit annoying to deal with in some cases due to the whole process title thingamajig.

`pidof` works great since it's also looking for the name of the process, not anything else (the exact behavior we need here!) I'd also bet on it being faster.

If you're worried about this function getting rather long, I bet you could move some of it into `get_procs()` and use the same method as Windows.

Reply:
Also if you'd like I can test using my real mac, though it's been tampered with by a bunch of development tools and by Homebrew.

Reply:
@jvyden I think it's fine to split macos and linux code, will you push the changes?

Reply:
Sure, I will get to this at some point, been busy lately haha